packages/compose/src/hooks/use-drop-zone/index.js

```diff
@@ -165,10 +181,22 @@ export default function useDropZone( {
 				// zone.
 				// Note: This is not entirely reliable in Safari due to this bug
 				// https://bugs.webkit.org/show_bug.cgi?id=66547
+
 				if ( isElementInZone( event.relatedTarget ) ) {
 					return;
 				}
 
+				// If we're moving in/out of a ZoomOutSeparator, don't trigger
+				// the onDragLeave event. This is to prevent the dropzone from
+				// being hidden when the user is dragging a block in/over/around
+				// a block and the separator.
```

Comment:
As this code is within `@wordpress/compose` we cannot refer to "ZoomOut". Therefore need to make this logic abstract such that it can be applied generically to the ZoomOutSeparator.